lib/web_ui/lib/src/engine/keyboard_binding.dart

```diff
@@ -152,6 +152,35 @@ class KeyboardBinding {
     _converter = KeyboardConverter(_onKeyData, onMacOs: operatingSystem == OperatingSystem.macOs);
   }
 
+  // Synthesize shift key up or down event only when the known pressing state is different.
+  void synthesizeShiftKeyIfNeeded(ui.KeyEventType type) {
+    // TODO(bleroux): should we take care of shift left AND shift right?
```

Comment:
Yes.

Let's use a "least event dispatched" strategy:
- If the true state is pressed, and neither key is pressed, press the left key.
- If the true state is not pressed, and any key is pressed, release these keys.
- Otherwise, dispatch no events.

Reply:
Done